### PR TITLE
Move teacher feedback components into sub-directory

### PR DESCRIPTION
**Pull request by maureensturgeon**
- Creates a new directory in `instructions` called `teacherFeedback` which contains all the teacher feedback components for the feedback tab of `TopInstructions`. 
- Renames of components to remove `TeacherFeedback-` and `Feedback-` prefixes which are no longer necessary with the directory context
- Moves types needed for teacher feedback tab into a new file within `teacherFeedback` directory

No functional changes

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jamescodeorg** [comment]
Could you remind me what we're renaming this tab to in the UI?

Do any of the components assume that they're used within TopInstructions?

**Comment by maureensturgeon** [comment]
> Could you remind me what we're renaming this tab to in the UI?
> 
> Do any of the components assume that they're used within TopInstructions?

The tab will be called Feedback unless there is a mini-rubric in which case it will be called Rubric. All of the components that are in the `instructions/teacherFeedback` directory are only used in the instructions component. Can you explain what you mean by "the components assume that they're used within TopInstructions"?

**Comment by jamescodeorg** [comment]
I was wondering if it would make sense to put `teacherFeedback` right under `templates` instead of `instructions` so I was curious how coupled these components are to TopInstructions.  I don't think I'll have a strong opinion either way so up to you.

**Comment by maureensturgeon** [comment]
> I was wondering if it would make sense to put `teacherFeedback` right under `templates` instead of `instructions` so I was curious how coupled these components are to TopInstructions. I don't think I'll have a strong opinion either way so up to you.

Since these are all only specifically used in the instructions right now and I don't foresee these being used anywhere else any time soon, I think it makes sense to nest them there

**Review by cforkish — APPROVED** [review]

File: apps/src/templates/instructions/teacherFeedback/TeacherFeedback.jsx [hunk 1]
```diff
@@ -280,7 +280,7 @@ export class TeacherFeedback extends Component {
             {viewAs === ViewType.Student &&
               this.renderCommentAreaHeaderForStudent()}
             {displayComment && (
-              <CommentArea
+              <Comment
```
Comment (jamescodeorg) [inline apps/src/templates/instructions/teacherFeedback/TeacherFeedback.jsx:283]: (unrelated) Is the `studentHasFeedback` property being used?

File: apps/src/templates/instructions/teacherFeedback/Status.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import color from '@cdo/apps/util/color';
 import i18n from '@cdo/locale';
 import {ViewType} from '@cdo/apps/code-studio/viewAsRedux';
 
-class TeacherFeedbackStatus extends Component {
+class Status extends Component {
```
Comment (jamescodeorg) [inline apps/src/templates/instructions/teacherFeedback/Status.jsx:9]: (nit) `Status` almost seems too short and generic.  Maybe `FeedbackStatus`?

File: apps/src/templates/instructions/teacherFeedback/ReviewState.jsx [hunk 3]
```diff
@@ -2,10 +2,10 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import i18n from '@cdo/locale';
 import {KeepWorkingBadge} from '@cdo/apps/templates/progress/BubbleBadge';
-import {ReviewStates} from '@cdo/apps/templates/types';
+import {ReviewStates} from '@cdo/apps/templates/instructions/teacherFeedback/types';
 import color from '@cdo/apps/util/color';
 
-class FeedbackStudentReviewState extends React.Component {
+class ReviewState extends React.Component {
```
Comment (jamescodeorg) [inline apps/src/templates/instructions/teacherFeedback/ReviewState.jsx:8]: (minor) `Maybe call this one ReadOnlyReviewState` for symmetry?  Or maybe `ReviewStateTeacher` and `ReviewStateStudent` would be clearer?

File: apps/src/templates/instructions/teacherFeedback/TeacherFeedback.jsx (hunk 1 above)
Comment (maureensturgeon) [reply apps/src/templates/instructions/teacherFeedback/TeacherFeedback.jsx:283]: Nope! Good catch

**Pull request merged**